File: frontend/frontend/src/app/services/stats-service.ts
import { Injectable } from '@angular/core';
import { StatsOverviewDto } from '../common/stats/stats-overview-dto';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root',
})
export class StatsService {
  private readonly baseUrl = 'http://localhost:8081/api/stats';

  constructor(private httpClient: HttpClient) {}

  getOverview(): Observable<StatsOverviewDto> {
    return this.httpClient.get<StatsOverviewDto>(`${this.baseUrl}/overview`);
  }
}

File: backend/src/main/java/com/strongBeton/strongBeton/service/StatsService.java
package com.strongBeton.strongBeton.service;

import com.strongBeton.strongBeton.dto.stats.StatsOverviewDTO;
import com.strongBeton.strongBeton.entity.user.User;

public interface StatsService {
    StatsOverviewDTO getStatsOverview(User currentUser);
}
